db/migrate/20220408103851_add_sender_institution_to_transfer_packages.rb

```diff
@@ -0,0 +1,18 @@
+class AddSenderInstitutionToTransferPackages < ActiveRecord::Migration
+  def up
+    add_reference :transfer_packages, :sender_institution, null: true
+
+    # Fill sender_institution on existing packages
+    TransferPackage.find_each do |package|
+      package.update_columns(
+        sender_institution_id: package.sample_transfers.first.sender_institution_id,
+      )
+    end
```

Comment:
Using models inside migration is often a gamble. For this reason I'm trying more and more to use only SQL for data migrations.

You're adding a column, so you must tell ActiveRecord to reload the model's definition from the database before using it, because Rails' may have already load it (especially in production).

Quick performance optimization: `#take` is better than `#first` to pick whatever row (it avoids an `ORDER BY`).

This is not worth it for this specific migration, but for learning purposes, I'm wondering if we could write this as a simple SQL request? With PostgreSQL we could (using `WITH` or something), but with MySQL 5.7? :thinking:

Reply:
Answering myself: it looks like this is possible with MySQL 8.0 :smile: but not before :disappointed: